AspNetSerilog/Extractors/HttpContextExtractor.cs

```diff
@@ -293,19 +293,53 @@ public static object GetPort(this HttpContext context)
         /// </summary>
         /// <param name="context"></param>
         /// <returns></returns>
-        public static object GetFullUrl(this HttpContext context, string[] blacklist)
+        public static object GetFullUrl(this HttpContext context, string[] queryBlacklist, string[] httpContextBlackList)
         {
             var absoluteUri = string.Concat(
                        context?.Request?.Scheme,
                        "://",
                        context?.Request?.Host.ToUriComponent(),
-                       context?.Request?.PathBase.ToUriComponent(),
-                       context?.Request?.Path.ToUriComponent(),
-                       context.GetRawQueryString(blacklist));
+                       context?.GetPathBase(httpContextBlackList),
+                       context?.GetPath(httpContextBlackList),
+                       context.GetRawQueryString(queryBlacklist));
 
             return absoluteUri;
         }
 
+        /// <summary>
+        /// Get Path
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public static object GetPath(this HttpContext context, string[] blacklist)
```

Comment:
O tipo de retorno aqui não vai ter nenhum impacto na atribuição do valor ?

Retornar um PathString não seria melhor ?

Reply:
Como o retorno já possui o ToUriComponent(), não tem impacto retornar dessa maneira. Ainda assim acha melhor tipar o retorno do método para string (que é o tipo retornado atualmente)?

Reply:
O tipo do Request.Path pelo que vi é `PathString`, então pensei do método `GetPath()` retornar assim e o método `ToUriComponent()` continuar sendo aplicado somente onde for necessário.